src/smtencoding/FStar.SMTEncoding.Z3.fs

```diff
@@ -497,28 +497,6 @@ let refresh () =
         bg_z3_proc.refresh();
         bg_scope := List.flatten (List.rev !fresh_scope)
 
-//mark, reset_mark, commit_mark:
-//    setting rollback points for the interactive mode
-// JP: I suspect the expected usage for the interactive mode is as follows:
-// - the stack (fresh_scope) has size >= 1, the top scope contains the queries
-//   that have been successful so far
-// - one calls "mark" to push a new scope of tentative queries
-// - in case of success, the new scope is collapsed with the previous scope,
-//   effectively bringing the new queries into the scope of successful queries so far
-// - in case of failure, the new scope is discarded
-let mark msg =
-    push msg
-let reset_mark msg =
-    // JP: pop_context (in universal.fs) does the same thing: it calls pop,
-    // followed by refresh
-    pop msg;
-    refresh ()
-let commit_mark (msg:string) =
```

Comment:
I'm confused. Before (assuming lists of strings for the sake of example):
```
l = []
push
l = [[]]
givez3 blah
l = [ [ blah ] ]
push
l = [ []; [blah] ]
givez3 bleh
l = [ [bleh]; [blah] ]

before:
commit
l = [ [ bleh; blah ] ]

now (commit is gone, only pop remains):
pop
l = [ [ blah ] ]
```

so your commit changes the behavior for fresh_scope, doesn't it?

Reply:
Yes and no. The trick is that it was not push, commit ; it was push, then mark (ie a second push), then commit. The commit merged the result of the mark with that of the push, so it went from [blah]; []; [bleh] to [blah]; [bleh]. Except it was worse, because the list that push added wasn't empty, so it went from ...;[Caption;Push];[Caption;Push;bleh] to [Caption;Push;Caption;Push;bleh]. I'm assuming that was a bug; wdyt?

Reply:
I chatted with @msprotz offline to confirm the description above; there's more details in the commit messages :)